### PR TITLE
feat(receipt-gate): CI gate blocks merge without PASS receipts [OMN-9051]

### PR DESCRIPTION
**Pull request by jonahgabriel**
[skip-receipt-gate: bootstrap — this PR adds the receipt-gate enforcement itself; first receipts produced on next PR post-merge]

## Summary

**Principle** (per user direction 2026-04-17 evening): "Enforcement by receipt. Ticket contracts and receipt and proof contracts equals the same fucking thing." Ticket contracts declare intent. Receipts declare fact. A PR does not merge unless every `dod_evidence` check on every cited ticket has a corresponding PASS receipt at the canonical path.

Closes the "aspirational enforcement" class of failure surfaced by tonight's dogfood audit (OMN-9062 validator unwired, OMN-9065 default not flipped, OMN-9051 spec inert — all shipped Done with CI green).

## What lands

- `EnumReceiptStatus` (PASS | FAIL)
- `ModelDodReceipt` — frozen Pydantic receipt schema
- `ModelReceiptCheckResult` + `ModelReceiptGateResult` (split per single-class-per-file)
- `omnibase_core.validation.receipt_gate.validate_pr_receipts` — full decision matrix
- `omnibase_core.validation.receipt_gate_cli` — \`python -m\` entrypoint
- `scripts/validation/validate_pr_receipts.py` — thin CLI wrapper
- `.github/workflows/receipt-gate.yml` — reusable workflow (workflow_call)
- pyproject.toml + .yaml-validation-allowlist.yaml allowlist entries

## Canonical paths

- Contracts: `<contracts-dir>/<OMN-XXXX>.yaml` (default `onex_change_control/contracts`)
- Receipts: `<receipts-dir>/<OMN-XXXX>/<evidence-item-id>/<check-type>.yaml` (default `onex_change_control/drift/dod_receipts`)

## Decision matrix

| Condition | Outcome |
|---|---|
| No ticket ref in PR body | FAIL |
| Ticket has no contract | FAIL |
| Contract has no dod_evidence | FAIL |
| Receipt missing for any check | FAIL |
| Receipt file unparseable | FAIL |
| Receipt schema invalid | FAIL |
| Receipt path↔content mismatch | FAIL |
| Receipt status != PASS | FAIL |
| `[skip-receipt-gate: <reason>]` override | PASS with friction |
| All checks have PASS receipts | PASS |

## Test plan

- [x] 27 tests on `ModelDodReceipt` (construction, frozen, extra-forbid, ticket_id regex, commit_sha hex regex, tz-aware timestamp, non-UTC normalization, negative duration, enum status)
- [x] 14 tests on receipt-gate decision matrix (no-ticket, missing contract, empty dod_evidence, corrupt contract, all-PASS, missing receipt, failing receipt, corrupt receipt, path-mismatch, multi-ticket any-fail, multi-ticket all-pass, override accepted, empty-reason-does-not-override)
- [x] All 41 tests green locally (\`uv run pytest\`)
- [x] ruff format + check clean
- [x] pre-commit (validate-string-versions, onex-single-class-per-file, validate-no-manual-yaml, topic-lint, secret-detection, all) clean on changed files
- [ ] CI (this PR)

## Follow-up tickets (to file as epic children)

1. Caller workflow in each of 12 repos + branch-protection required status check
2. \`node_dod_verify\` emits real receipts instead of ad-hoc YAML (OMN-9074)
3. Receipt emitters in existing validators (OMN-9113 ValidatorBannedComposeVars, etc.)
4. Dogfood-sweep tick that audits closed tickets for receipt presence (auto-reopens on missing)
5. Unify \`ModelTicketContract\` spec across repos (onex_change_control vs omnibase_core.models.contracts.ticket.model_linear_ticket_contract)

## Rationale

Tonight's dogfood audit found that 3 of the 7 "enforcement surfaces" shipped today were paper (code merged, CI green, ticket Done, but the enforcement never actually executes on a real PR). The existing gates enforce **process** (CR threads resolved, contract present, deploy ticket cited) but not **function** (did the validator actually run, did the default actually take effect, did the consumer actually read the spec).

Receipts close that gap: the only way to PASS this gate is a file on disk asserting the check ran with a recorded outcome. Absence = FAIL. Status != PASS = FAIL.

The override exists (`[skip-receipt-gate: <reason>]`) for emergency hotfixes but logs friction — every override must be tracked and closed within 7 days.

## Summary by CodeRabbit

* **New Features**
  * Add a reusable CI workflow and CLI to validate PR-linked receipts (configurable contracts/receipts paths); emits workflow annotations and can block PRs.
  * Introduce PASS/FAIL receipt status, structured receipt and gate-result formats, and a PR-body override tag to skip validation when a non-empty reason is supplied.
* **Tests**
  * Comprehensive unit tests covering parsing, schema validation, multi-ticket scenarios, override behavior, and edge cases.

[skip-deploy-gate: receipt-gate meta-PR landing the enforcement workflow itself; no deploy-agent action required]

### CODE REVIEW

File: .github/workflows/receipt-gate.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+          python-version: "3.12"
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+
+      - name: Install omnibase_core
+        run: |
+          uv pip install --system omnibase-core || \
+            uv pip install --system -e ./omnibase_core || \
+            echo "::warning::omnibase_core install path not found; falling back to source checkout"
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply .github/workflows/receipt-gate.yml]: Deferring — install-failure fast-fail is a valid hygiene improvement but not load-bearing for the gate's correctness. The actual receipt-gate step will fail loudly if the package isn't importable; resolving to unblock and tracking as hygiene.

File: .github/workflows/receipt-gate.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+      - name: Run Receipt-Gate
+        env:
+          PR_BODY: ${{ github.event.pull_request.body }}
+        run: |
+          python -m omnibase_core.validation.receipt_gate_cli \
+            --pr-body "$PR_BODY" \
+            --contracts-dir ".onex_change_control/${{ inputs.contracts-dir }}" \
+            --receipts-dir ".onex_change_control/${{ inputs.receipts-dir }}"
```
[the remark this answers is not in the document]
Comment (jonahgabriel) [reply .github/workflows/receipt-gate.yml]: Deferring — current double-quoting handles the vast majority of PR body content; heredoc/tempfile refactor is a defensible hardening but not a blocker. Tracking for hygiene pass.
